popgen/AFSsimulate.go

```diff
@@ -2,14 +2,89 @@ package popgen
 
 import (
 	"github.com/vertgenlab/gonomics/numbers"
+	"github.com/vertgenlab/gonomics/vcf"
+	"math/rand"
+	//DEBUG: "fmt"
 )
+
 //SimulateAFS returns an allele frequency spectrum AFS struct for n individuals with k segregating sites
 //with a selection parameter alpha.
-/*func SimulateAFS(alpha float64, n int, k int) AFS {
+func SimulateAFS(alpha float64, n int, k int) AFS {
 	var answer AFS
 	answer.sites = make([]*SegSite, 0)
+	var alleleFrequencies []float64 = StationaritySampler(alpha, k, 100000, 1000, 0.001, 0.999, true)
+	var count int
+	var r float64
 	for i := 0; i < k; i++ {
-		answer.sites = append(answer.sites, &SegSite{StationaritySample(n, alpha), n})
+		//now we simulate the discrete number of alleles for n individuals based on the allele frequency
+		count = 0
+		for j := 0; j < n; j++ {
+			r = rand.Float64()
+			if r < alleleFrequencies[i] {
+				count++ 
+			}
+		}
+		answer.sites = append(answer.sites, &SegSite{count, n})
+	}
+	return answer
+}
+
+//SimulateGenotype creates a matrix of GenomeSamples with k rows and n/2 columns, corresponding to k segregating sites and n alleles.
+//based on a selection parameter alpha. Alleles are shuffled among the individuals.
+func SimulateGenotype(alpha float64, n int, k int) [][]vcf.GenomeSample {
+	a := SimulateAFS(alpha, n, k)
+	var answer [][]vcf.GenomeSample = make([][]vcf.GenomeSample, 0, k)
+	
+	//now we initialize all column entries for each row of the answer matrix
+	for i := 0; i < len(answer); i++ {
```

Comment:
I don't think this loop helps you because you are not doing: answer[i] = append(answer[i], something)
Later on you run: answer = append(answer[i], something)
I think you can comment out this loop because it is allocating memory that is then not being used.